File: Array Leaders.py
class Solution:
    def leaders(self, arr):
        # code here
        L=[]
        max_a=float("-Inf")
        
        for i in range(len(arr)-1,-1,-1):
            if max_a <= arr[i]:
                L.append(arr[i])
                max_a=arr[i]
        L.reverse()
        return L

File: Array Leaders.js
// User function Template for javascript

/**
 * @param {number[]} a
 * @returns {number[]}
 */
class Solution {
    // Function to find the leaders in the array.
    leaders(arr) {
        // code here
        let r= []
        let max_a= -Infinity
        for (let i= arr.length-1; i>=0; i--){
            if (max_a <= arr[i]){
                r.push(arr[i])
                max_a=arr[i]
            }
        }
        r.reverse()
        return r
    }
}
